### DISCUSSION TITLE
Transaction reverted on lesson 9 NFT

### DISCUSSION THREAD

**Discussion by m4k2** [Cyfrin/foundry-full-course-cu#278] [Q&A]
Hey, is it typical for every transaction to fail on the Sepolia Lesson 9 NFT challenge? You can check the transaction history at https://sepolia.etherscan.io/txs?a=0x33e1fd270599188bb1489a169df1f0be08b83509.

**Comment by shikhar229169** [comment]
Actually when you interact with a contract via another contract, the txn will not show up in the latter contract.
Suppose I interacted with A contract, and A contract sent a txn to B contract. So what will happen, the txn will show up in A contract, but will not show up in B contract.
I think you will get the idea by this.

**Comment by m4k2** [comment]
Okay do you know why my script keep failing tho ? i know i send the right value for `randomGuess` because i don't get the error `error LessonNine__WrongValue();`. 
Furthermore before every script execute, an onchain simulation is done, and i passed it each time so i really got the right value of `randomGuess`.

this is the output of my running script : 
```bash 
[⠒] Compiling...
No files changed, compilation skipped
Script ran successfully.

## Setting up (1) EVMs.

==========================

Chain 11155111

Estimated gas price: 3.000000024 gwei

Estimated total gas used for script: 195265

Estimated amount required: 0.00058579500468636 ETH

==========================

###
Finding wallets for all the necessary addresses...
##
Sending transactions [0 - 0].
⠁ [00:00:00] [#################################################] 1/1 txes (0.0s)
Transactions saved to: /home/mk/Desktop/CTF/PatricksCollins/Lesson_9/broadcast/Exploitoor.s.sol/11155111/run-latest.json

##
Waiting for receipts.

Transactions saved to: /home/mk/Desktop/CTF/PatricksCollins/Lesson_9/broadcast/Exploitoor.s.sol/11155111/run-latest.json

Error: 
Transaction Failure: 0xafc0a0efd2400092edb3161d673dc7c43fc689a747fba7bdc511936e4d4a4465
```

thanks for your help

**Reply by f3d0ss** [reply]
How are you doing the "onchain simulation"?
Because if you are using the foundry tests you have to consider the fact that you can reed the current `block.timstamp` and `block.randao` and execute all the transaction you want in that block.
But in a real enviroment you would read the `block.timstamp` and `block.randao` of the last block and propagate your new transaction that will be included in subsequent block (with different `block.timstamp` and `block.randao`)

**Reply by m4k2** [reply]
Every foundry script performs an on-chain simulation before execution. To verify this, you can set `randomGuess` to `0x00`. The script will fails to execute and reports a revert with the `error LessonNine__WrongValue()` selector, it means that the on-chain simulation has detected the transaction will revert ( Guess != Answer ).

**Comment by shikhar229169** [comment]
Can you please tell the method by which you are calculating the value of ```randomGuess```
According to me, it is due to passing of wrong value.

**Comment by m4k2** [comment]
Yeah got the answer to my problem here : https://github.com/Cyfrin/foundry-full-course-f23/discussions/115#discussioncomment-6327258 

I was trying to solve without a new smart contract. I can share my script and maybe we can find a way together ( or at less why the script failed ) to solve the lesson 9 without a smart contract.

**Reply by Kel-VinTech** [reply]
can i contact you?
